client/devdocs/design/search-collection.jsx

```diff
@@ -0,0 +1,98 @@
+/**
+* External dependencies
+*/
+import React from 'react';
+
+const Hider = React.createClass( {
+	displayName: 'Hider',
+
+	propTypes: {
+		hide: React.PropTypes.bool,
+	},
+
+	shouldComponentUpdate( nextProps ) {
+		return this.props.hide !== nextProps.hide;
+	},
+
+	render() {
+		return (
+			<div style={ this.props.hide ? { display: 'none' } : {} }>
+				{ this.props.children }
+			</div>
+		);
+	}
+} );
+
+const FilterSummary = React.createClass( {
+	render() {
+		let names;
+
+		if ( this.props.items.length === 0 ) {
+			return ( <p>No matches found</p> );
+		} else if ( this.props.items.length === this.props.total || this.props.items.length === 1 ) {
+			return null;
+		}
+
+		names = this.props.items.map( function( item ) {
+			return item.props.children.type.displayName;
+		} );
+
+		return (
+			<p>Showing: { names.join( ', ' ) }</p>
+		);
+	}
+} );
+
+export default React.createClass( {
+	displayName: 'Collection',
+
+	shouldWeHide: function( example ) {
+		let filter, searchString;
+
+		filter = this.props.filter || '';
+		searchString = example.type.displayName;
+
+		if ( this.props.component ) {
+			return example.type.displayName.toLowerCase() !== this.props.component.replace( /-([a-z])/g, '$1' );
+		}
+
+		if ( example.props.searchKeywords ) {
+			searchString += ' ' + example.props.searchKeywords;
+		}
+
+		return ! ( ! filter || searchString.toLowerCase().indexOf( filter ) > -1 );
+	},
+
+	visibleExamples: function( examples ) {
+		return examples.filter( ( child ) => {
```

Comment:
This will probably be an issue too if we were ever to render the component with non-array children. We're probably better off calling `React.Children.toArray` to force it to the form we're expecting. It's not an immediate concern, however, since it's not currently the case that we'd be rendering the component this way.